src/nevermined/Assets.ts

```diff
@@ -115,16 +146,47 @@ export class Assets extends Instantiable {
     }
 
     /**
-     * Returns a DDO by DID.
+     * Returns a DDO by DID. Depending of the resolution policy it prioritize the Metadata API or Immutable urls.
      * @param did - Decentralized ID.
+     * @param policy - It specifies the resolve policy to apply. It allows to select that priorities during the asset resolution via Metadata API or Immutable URLs (IPFS, Filecoin, etc)
      * @returns {@link DDO}
      */
-    public async resolve(did: string): Promise<DDO> {
-        const { serviceEndpoint } =
+    public async resolve(did: string, policy: DIDResolvePolicy = DIDResolvePolicy.ImmutableFirst): Promise<DDO> {
```

Comment:
I think the default should be `OnlyMetadataAPI` to be backwards compatible

Reply:
`ImmutableFirst` should be backwards compatible. In the existing DIDs the `immutableUrl` on-chain parameter will be empty so that policy will resolve always from the Marketplace API. 
Having `ImmutableFirst` by default prioritized immutable data stores in the stack using Marketplace API for searching.

Main concern is if in the long term using IPFS or similar to download DDOs have any drawback from a performance point of view.